### PR TITLE
fixed import code, removed module level imports

### PR DESCRIPTION
**Pull request by matt-chan**
Fixes issue #16

### CODE REVIEW

**Comment by tovrstra** [comment]
There seems to be the author problem again. I don't understand where it comes from now. Normally, your commits are created on your own computed, so author names should not be affected by github then?

**Comment by matt-chan** [comment]
They are. And my gitk shows the correct author name... Also I spoke with support and it looks like I'm doing everything correctly...

Is the error from the author problem? Travis seemed to report something about failed tests instead?

**Comment by matt-chan** [comment]
Okay. I've fixed the failing nosetest stuff... 

I'm still getting bad author though, which makes no sense to me... The git history is clean as far as I can see.

Do you mind looking into this a bit please Toon? It's not clear to me how the QA scripts check authors (where are the scripts anyways?)

**Comment by tovrstra** [comment]
I'll look into it. The script is `tools/qa/check_names.py`.

**Comment by matt-chan** [comment]
Thanks!

**Comment by tovrstra** [comment]
OK, this is github still making a mess out of your account. If you want to see the branch on which Travis runs the tests, you can do the following locally:

    git checkout master
    git checkout -b pull55
    git pull origin refs/pull/55/merge

This can be seen in the Travis log. This means that Travis gets to test on a preliminary merge of the pulled branch with the master branch. In that merge, they are sill using your anonymized e-mail, which is a github bug.

**Comment by matt-chan** [comment]
Okay, I think I found the change that she was talking about. The email should be fixed now.

Is there a way to test this again? I could probably just recreate the PR, but does Travis have a retest feature?

**Comment by tovrstra** [comment]
Great!

That should work. Another option is to do a `git commit --amend` and force push that to your feature branch. (Not as nice but quick.)

**Comment by tovrstra** [comment]
It all looks good now. That's a relief. Thanks!

**Comment by tovrstra** [comment]
I checked the diff and I don't see any problems. Can be merged as soon as tests pass.

**Comment by tovrstra** [comment]
It may be worth while to first merge PR #58 because that will make it easier to see which errors are the show stoppers.

**Comment by tovrstra** [comment]
I could also recommend the following: one can locally quickly run a trapdoor script to see what the result will be. Especially for pep8 and pep257, that goes very quickly.

For the first time run the following:

    ./tools/qa/trapdoor_pep8.py feature
    cp tools/qa/trapdoor* qaworkdir/
    git checkout master
    ./qaworkdir/trapdoor_pep8.py ancestor
    git checkout feature
    ./qaworkdir/trapdoor_pep8.py report

To get a more detailed report:

    ./qaworkdir/trapdoor_pep8.py report -n

To limit the messages to a given file:

    ./qaworkdir/trapdoor_pep8.py report -n -f wpart.py

When continuing work on the feature branch, it is sufficient to just run the following for an update of the messages:

    ./tools/qa/trapdoor_pep8.py feature
    ./qaworkdir/trapdoor_pep8.py report

The report output will become more useful with PR #58, but even at this stage it is useful.

**Comment by matt-chan** [comment]
I had some problems trying to run pep257 checks. The binary isn't in the fedora 23 repos I think?

**Comment by tovrstra** [comment]
pep257 is not in the fedora 23 (nor in 24). It is better

File: horton/scripts/wpart.py [hunk 1]
```diff
@@ -119,7 +119,8 @@ def wpart_slow_analysis(wpart, mol):
     operators = []
     for ipure in xrange(npure):
         for iatom in xrange(wpart.natom):
-            operators.append(wpart.cache.load('overlap_operators', iatom)['olp_%05i' % ipure])
+            operators.append(wpart.cache.load('overlap_operators', iatom)[
+                                 'olp_%05i' % ipure])
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: C0330 bad-continuation Wrong hanging indentation (remove 17 spaces).

File: horton/scripts/wpart.py [hunk 2]
```diff
@@ -17,9 +17,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, see <http://www.gnu.org/licenses/>
 #
-#--
-'''Utility functions for the ``horton-wpart.py`` script'''
-
+# --
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: Can you undo this change? This type of errors should all be fixed at once, together with a fix in the script `updateheaders.sh`. See #17. This pep8 error should not be causing problems.

File: horton/scripts/wpart.py [hunk 3]
```diff
@@ -77,25 +76,26 @@ def wpart_slow_analysis(wpart, mol):
         # Prepare solid harmonics on grids.
         grid = wpart.get_grid(index)
         if wpart.lmax > 0:
-            work = np.zeros((grid.size, npure-1), float)
-            work[:,0] = grid.points[:,2] - wpart.coordinates[index,2]
-            work[:,1] = grid.points[:,0] - wpart.coordinates[index,0]
-            work[:,2] = grid.points[:,1] - wpart.coordinates[index,1]
+            work = np.zeros((grid.size, npure - 1), float)
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: Just minor note: adding whitespace around arithmetic operators is not a pep8 requirement. It is apparently not a generally accepted style (for some languages). See http://stackoverflow.com/questions/18171560/python-coding-style-is-whitespace-around-operator-needed-in-functions-actual-p

File: horton/scripts/wpart.py [hunk 4]
```diff
@@ -161,7 +164,8 @@ def wpart_slow_analysis(wpart, mol):
     if dm_spin is None:
         if not hasattr(mol, 'exp_alpha'):
             return
-        xs_response = 2*compute_noninteracting_response(mol.exp_alpha, operators)
+        xs_response = 2 * compute_noninteracting_response(mol.exp_alpha,
+                                                          operators)
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: Just a minor note: We've set the maximum line length to 100. It is a good idea to set your editor a little more conservativelye, e.g. 90, but we found 80 a little too harsh.

**Comment by tovrstra** [comment]
The counter for pylint error E1136 should not have increased. I'll try to figure this out. It is a false positive anyway.

**Comment by tovrstra** [comment]
The trapdoor scripts fail to find the correct ancestor branch to compare to. The reason is that travis runs the tests on a preliminary merge of the pull request and the master, making it hard to find the correct ancestor commit. I've always tested these scripts with the assumption that the feature branch was not merged yet. hmmm. I'll see what I can do.

You could fix this by rebasing but please don't do that yet. This is a good test case for fixing the qa scripts.

**Comment by tovrstra** [comment]
I took a closer look: the ancestor is indeed different but actually not causing the problem. The real problem is apparently an obscure pylint bug. It depends on how you run Pylint whether the error E1136 is reported. I would suggest to add E1136 to the list of exceptions in `tools/qa/pylintrc` (line 62) and list it also in the documentation in `doc/tech_dev_checklist.rst` (line 545) as one of the excluded checks due to false positives.

**Comment by tovrstra** [comment]
Can you also exclude W0621? I just saw that that also generates tons of false positives. Thanks.

File: horton/scripts/wpart.py (hunk 3 above)
Comment (matt-chan) [inline horton/scripts/wpart.py]: I would push for it though... Pycharm does it automatically when it's reformatting the file so if you want the rest of the auto formatting, you need to take it too. There might be a way to disable that particular auto-format, but it would mean every new developer using pycharm would have to configure it going forward...

Also it does look a bit cleaner?

Comment (tovrstra) [inline horton/scripts/wpart.py]: No problem. I'm fine either way. I was only worried that it would take extra efforts to add these spaces, but that is clearly not a concern. (It is rather the opposite.)

File: horton/scripts/wpart.py [hunk 5]
```diff
@@ -119,7 +119,8 @@ def wpart_slow_analysis(wpart, mol):
     operators = []
     for ipure in xrange(npure):
         for iatom in xrange(wpart.natom):
-            operators.append(wpart.cache.load('overlap_operators', iatom)['olp_%05i' % ipure])
+            operators.append(wpart.cache.load('overlap_operators', iatom)[
+                             'olp_%05i' % ipure])
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: From Travis: `C0330 Wrong hanging indentation (remove 13 spaces).`
I would do this differently that pylint suggests.
1) It is ok best to just put it just back on one line.
2) If you want to wrap, put the square bracket also on the second line, i.e. wrap after `iatom)` not after `iatom)[`.

Comment (tovrstra) [inline horton/scripts/wpart.py]: P.S. If this wrapping was done automatically by pycharm, you should be really careful with the automatical fixes. It would be unfortunate to make things worse that way.

**Comment by tovrstra** [comment]
This PR also fixes #90.

File: doc/tech_dev_checklist.rst [hunk 6]
```diff
@@ -386,7 +386,7 @@ The following messages are excluded by default: I0020, I0021, W0704. (It is not
 these stand for. They are not documented in Pylint.)
 
 The following messages are excluded by default but are activated in our case (related to Python
-3): E1601, E1602, E1603, E1604, E1605, E1606, E1607, E1608, W1601, W1602, W1603, W1604,
+3): E1136, E1601, E1602, E1603, E1604, E1605, E1606, E1607, E1608, W0621, W1601, W1602, W1603, W1604,
```
Comment (tovrstra) [inline doc/tech_dev_checklist.rst]: This is not fixed in the correct please. (These exceptions should be listed somewhat lower.)

File: horton/scripts/wpart.py [hunk 7]
```diff
@@ -17,10 +17,9 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, see <http://www.gnu.org/licenses/>
 #
-#--
+# --
```
Comment (tovrstra) [inline horton/scripts/wpart.py]: Please undo. It breaks the updateheaders.sh script.

Comment (tovrstra) [inline horton/scripts/wpart.py]: See #17. This will be fixed in a later release.

File: horton/scripts/wpart.py (hunk 5 above)
Comment (matt-chan) [inline horton/scripts/wpart.py]: Yeah, the line wrapping is sometimes a bit wonky. But I guess pylint will catch it if it breaks things. The weird thing is that pylint was still complaining after I justified it correctly (according to PEP anyways).

I've unrolled it back to one line now!

**Comment by tovrstra** [comment]
Looks good! Thanks! Merging...

**Comment by matt-chan** [comment]
Haha finally. 

Thanks Toon!
